content/blog/2022/2022-08-25-schedule-cron-job-in-node.md

```diff
@@ -0,0 +1,318 @@
+
+`output`
+```
+Application Listening.....
+---------------------
+running a task every 15 second
+```
+
+
+### 2. Scheduling Email Using Node Cron
+Emailing is a common feature of modern applications. Cron jobs can be used to accomplish this. For instance, a job schedule can be set to automatically send users an email each month with the most recent information from a blog or a product.
+
+Here we are using google's email service provider `gmail`, if you have a Gmail account insert it in the code below to test out our newly created email scheduler.
+
+Note: To use node-mailer with Gmail, you must first create an [app password](https://support.google.com/mail/answer/185833?hl=en-GB) for Gmail to allow third-party access.
+
+Set up your Gmail app password following these steps:
+- First head to your Gmail Account.
+- Click on the profile image to the right
+- Next click on `Manage your Google Account`, then click `Security`.
+- In the `Signing in to Google` section select `App password` option.
+- If the `App password` option is unavailable, set up 2-Step Verification for your account.
+- Select the app (mail) and the current device you want to generate the app password for.
+- Click Generate.
+- Copy the generated 16-character code in the yellow bar on your device.
+
+To create our email scheduler application Insert the following code into the `index.js` file:
+```javascript
+const express = require("express");
+const cron = require("node-cron");
+const nodemailer = require("nodemailer");
+app = express();
+
+//Send email after 1 minute
+cron.schedule("1 * * * *", function () {
+  mailService();
+});
+
+function mailService() {
+  let mailTransporter = nodemailer.createTransport({
+    service: "gmail",
+    auth: {
+      user: "<your-email>@gmail.com",
+// use generated app password for gmail
+      pass: "***********",
+    },
+  });
+
+  // Setting credentials
+  let mailDetails = {
+    from: "<your-email>@gmail.com",
+    to: "<user-email>@gmail.com",
+    subject: "Test Mail using Cron Job",
+    text: "Node.js Cron Job Email Demo Test from Reflectoring Blog",
+  };
+
+  // Sending Email
+  mailTransporter.sendMail(mailDetails, function (err, data) {
+    if (err) {
+      console.log("Error Occured", err.message);
+    } else {
+      console.log("---------------------");
+      console.log("Email sent successfully");
+    }
+  });
+}
+
+app.listen(3000, () => {
+  console.log("Application Listening.....");
+});
+
+```
+In the above code we are using the node-mailer and node-cron dependencies we installed earlier. The `node-mailer` dependency gives us the feature of sending e-mails from our Node.js applications, using any email service of your choice.
+
+We scheduled our mail to be sent at `At 1 minute` interval using Gmail.
+
+Run the script `node index.js`
+
+`output:`
+```
+Application Listening.....
+---------------------
+Email Sent Successfully
+```
+Check your inbox to confirm the email is sent.
+
+### 3. Writing to a Log File
```

Comment:
I would rename this to "Monitoring Server Resources Over Time" because that is the valuable use case here ... logging to a file is just the means we're using to get there.